### PR TITLE
Improves Medical Hud interface and functionality

### PR DESCRIPTION
**Pull request by DoutorWhite**
## About the PR

There are some changes to the Medical Hud interface: The icon has been changed and now there is one for each mob state, making it easier to recognize whether the mob is rotting, dead or crit.

## Why / Balance

It's more visually pleasing and helps with discerning mob states.

## Technical details

There is a decision structure for determining which icon is appropriate for the mob's state. It's the only relevant code change.

## Media

Captura de tela 2024-03-13 230527

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Introduce new health status icons.
- tweak: Enhance the Medical HUD to display the health status of players and mobs using the newly added icons.

### CODE REVIEW

**Comment by UbaserB** [comment]
I was thinking about the ss13 states, but these look good. Only thing i would say is there needs to be some distinction.. so the sprites themselves should be different in some way besides a colour change.

**Comment by DoutorWhite** [comment]
> I was thinking about the ss13 states, but these look good. Only thing i would say is there needs to be some distinction.. so the sprites themselves should be different in some way besides a colour change.

It can definitely be done! In its current state it's a bit of programmer art, I'm not very good at spriting.

File: Resources/Textures/Interface/Misc/health_icons.rsi/meta.json [hunk 1]
```diff
@@ -5,10 +5,19 @@
...
+        },
+		{
+			"name": "Critical"
+		},
+		{
+			"name": "Dead"
+		},
+		{
+			"name": "Catatonic"
+		}
```
Comment (ficcialfaint) [inline Resources/Textures/Interface/Misc/health_icons.rsi/meta.json:21]: Use whitespaces not tabs

Comment (DoutorWhite) [reply Resources/Textures/Interface/Misc/health_icons.rsi/meta.json]: Solved

**Comment by UbaserB** [comment]
> > I was thinking about the ss13 states, but these look good. Only thing i would say is there needs to be some distinction.. so the sprites themselves should be different in some way besides a colour change.
> 
> It can definitely be done! In its current state it's a bit of programmer art, I'm not very good at spriting.

take the /tg/ sprites iirc

**Comment by DoutorWhite** [comment]
> iirc



> > > I was thinking about the ss13 states, but these look good. Only thing i would say is there needs to be some distinction.. so the sprites themselves should be different in some way besides a colour change.
> > 
> > 
> > It can definitely be done! In its current state it's a bit of programmer art, I'm not very good at spriting.
> 
> take the /tg/ sprites iirc

Looks better?

**Comment by UbaserB** [comment]
> > iirc
> 
> > > > I was thinking about the ss13 states, but these look good. Only thing i would say is there needs to be some distinction.. so the sprites themselves should be different in some way besides a colour change.
> > > 
> > > 
> > > It can definitely be done! In its current state it's a bit of programmer art, I'm not very good at spriting.
> > 
> > 
> > take the /tg/ sprites iirc
> 
> Looks better?

the rounded critical state icon is a bit weird, but for the most part yes

**Comment by DoutorWhite** [comment]
> > > iirc
> > 
> > 
> > > > > I was thinking about the ss13 states, but these look good. Only thing i would say is there needs to be some distinction.. so the sprites themselves should be different in some way besides a colour change.
> > > > 
> > > > 
> > > > It can definitely be done! In its current state it's a bit of programmer art, I'm not very good at spriting.
> > > 
> > > 
> > > take the /tg/ sprites iirc
> > 
> > 
> > Looks better?
> 
> the rounded critical state icon is a bit weird, but for the most part yes

I left it with the entire border and increased the contrast, but I'm not going to update the image now because I'm sleepy, honestly. Thanks

File: Resources/Textures/Interface/Misc/health_icons.rsi/meta.json [hunk 2]
```diff
@@ -5,10 +5,29 @@
         "y": 8
     },
     "license": "CC-BY-SA-3.0",
-    "copyright": "https://github.com/tgstation/tgstation/blob/master/icons/mob/huds/hud.dmi",
+    "copyright": "Taken from https://github.com/tgstation/tgstation/commits/master/icons/mob/huds/hud.dmi and modified by DoutorWhite",
```
Comment (metalgearsloth) [inline Resources/Textures/Interface/Misc/health_icons.rsi/meta.json:8]: This needs to reference the commit it was taken from.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Thought I reviewed this but apparently not.

The health icons should be tied to mobstates via dictionary or array rather than each one having its own hardcoded string and a big if-else branch.

File: Resources/Textures/Interface/Misc/health_icons.rsi/meta.json (hunk 2 above)
Comment (DoutorWhite) [reply Resources/Textures/Interface/Misc/health_icons.rsi/meta.json]: I think I solved both issues

**Review by EmoGarbage404 — DISMISSED** [review]
i feel like these sprites are kinda out of style and i'd rather just use the ones from SS13.

**Comment by DoutorWhite** [comment]
> i feel like these sprites are kinda out of style and i'd rather just use the ones from SS13.

These are all sprites from 13, I just 

> i feel like these sprites are kinda out of style and i'd rather just use the ones from SS13.

I mean, these sprites were taken from 13, with recolor it to make it less ambiguous.

**Comment by DoutorWhite** [comment]
Sorry for the request, PJB, I moved to a random branch to force a refresh on the PR because of a bug.

**Comment by DoutorWhite** [comment]
Why is GitHub like this?

**Comment by DoutorWhite** [comment]
Explaining what happened: I tried to force GitHub to update the PR and in the process it became extremely dirty and requested several code owners reviews.

**Comment by Golinth** [comment]
> > i feel like these sprites are kinda out of style and i'd rather just use the ones from SS13.
> 
> I mean, these sprites were taken from 13, with recolor it to make it less ambiguous.

Maybe use the most recent TG station ones? From [this commit](https://github.com/tgstation/tgstation/pull/81681/commits/c3fef620cc7117e53f0e5d0bbcde62b13442334a)

**Comment by DoutorWhite** [comment]
> > > i feel like these sprites are kinda out of style and i'd rather just use the ones from SS13.
> > 
> > 
> > I mean, these sprites were taken from 13, with recolor it to make it less ambiguous.
> 
> Maybe use the most recent TG station ones without the recolor? From the commit you already have linked

Yes! I already changed it.

**Comment by metalgearsloth** [comment]
> Thought I reviewed this but apparently not.
> 
> The health icons should be tied to mobstates via dictionary or array rather than each one having its own hardcoded string and a big if-else branch.

okay I had in fact reviewed it so https://github.com/space-wizards/space-station-14/pull/25711

**Comment by DoutorWhite** [comment]
> > Thought I reviewed this but apparently not.
> > The health icons should be tied to mobstates via dictionary or array rather than each one having its own hardcoded string and a big if-else branch.
> 
> okay I had in fact reviewed it so #25711

Very similar, in fact, I'm actually impressed. Honestly, I had no idea this PR existed, otherwise I wouldn't have even opened it. It was like Newton and Leibniz inventing differential calculus

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**